### PR TITLE
cephfs-shell: fix rmdir cmd

### PR DESCRIPTION
**Pull request by dparmar18**
Description: rmdir cmd's description bettered, made rmdir cmd's error 
output similar to bash's rmdir error output.

Fixes: https://tracker.ceph.com/issues/40860

Signed-off-by: Dhairya Parmar<dparmar@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by dparmar18** [comment]
> > Tested!!Any special reason for `directory` in block letters.Otherwise LGTM
> 
> Oops, changing it now

@neesingh-rh done

File: src/tools/cephfs/cephfs-shell [hunk 1]
```diff
@@ -882,6 +883,9 @@ class CephFSShell(Cmd):
         """
         Remove a specific Directory
         """
+        errors = {2: 'No such file or directory',
+                  20: 'Not a directory',
+                  39: 'Directory not empty'}
```
Comment (rishabh-d-dave) [inline src/tools/cephfs/cephfs-shell:888]: I have same opinion here, better to not add many mappings of error code to error message and rather deal with problem in a single go.

File: src/tools/cephfs/cephfs-shell [hunk 2]
```diff
@@ -865,14 +865,15 @@ class CephFSShell(Cmd):
         """
         return self.complete_filenames(text, line, begidx, endidx)
 
-    rmdir_parser = argparse.ArgumentParser(description='Remove Directory.')
-    rmdir_parser.add_argument('paths', help='Directory Path.', nargs='+',
+    rmdir_parser = argparse.ArgumentParser(description='Remove the '
+                                                       'directory(ies), '
+                                                       'if they are empty.')
```
Comment (rishabh-d-dave) [inline src/tools/cephfs/cephfs-shell:870]: nit: better to move description to next line and fit it in a single line?

Comment (rishabh-d-dave) [inline src/tools/cephfs/cephfs-shell:870]: nit: better to move description to next line and fit it in a single line?

**Review by rishabh-d-dave — CHANGES_REQUESTED** [review]
Looks good otherwise.

File: src/tools/cephfs/cephfs-shell [hunk 3]
```diff
@@ -922,7 +926,12 @@ class CephFSShell(Cmd):
                 try:
                     cephfs.rmdir(path)
                 except libcephfs.Error as e:
-                    set_exit_code_msg(msg=e)
+                    if e.get_error_code() in errors.keys():
+                        set_exit_code_msg(msg=f"rmdir: failed to remove "
+                                              f"{path.decode('utf-8')}: "
+                                              f"{errors[e.get_error_code()]}")
```
Comment (rishabh-d-dave) [inline src/tools/cephfs/cephfs-shell:932]: A `str` type is being passed to to `set_exit_code_msg()`. This sets the exit code for cephfs-shell to 1 which is `Misc`, ignoring `exit_codes` dict.

**Review by rishabh-d-dave — CHANGES_REQUESTED** [review]

File: src/tools/cephfs/cephfs-shell (hunk 3 above)
Comment (dparmar18) [reply src/tools/cephfs/cephfs-shell]: Done, thanks

File: src/tools/cephfs/cephfs-shell (hunk 1 above)
Comment (dparmar18) [reply src/tools/cephfs/cephfs-shell]: Same as #46185, removed. Thanks

File: src/tools/cephfs/cephfs-shell (hunk 2 above)
Comment (dparmar18) [reply src/tools/cephfs/cephfs-shell]: Done

**Review by rishabh-d-dave — APPROVED** [review]

**Comment by rishabh-d-dave** [comment]
QA run was successful - http://pulpito.front.sepia.ceph.com/rishabh-2022-05-20_05:29:37-fs-wip-rishabh-shell-rm-distro-basic-smithi/

**Comment by rishabh-d-dave** [comment]
This PR must wait for PR #46337 to merge though.

**Comment by vshankar** [comment]
> This PR must wait for PR #46337 to merge first though.

merged.

**Pull request merged**